src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Sdk.Before.targets

```diff
@@ -21,4 +23,66 @@
+    <!-- Windows has 2 properties that could be set, so if only 1 is set we copy it to the other. -->
+    <TargetPlatformMinVersion   Condition="'$(TargetPlatformMinVersion)'  ==  '' and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">$(SupportedOSPlatformVersion)</TargetPlatformMinVersion>
+    <SupportedOSPlatformVersion Condition="'$(SupportedOSPlatformVersion)' == '' and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">$(TargetPlatformMinVersion)</SupportedOSPlatformVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetFramework)' != '' and '$(MauiDisableSupportedOSPlatformVersionValidation)' != 'true' and '$(MauiDisableOverrideSupportedOSPlatformVersion)' != 'true'">
+    <!-- Record it the explicit version was lower than what we support so we can increase and log later. -->
+    <_MauiHadToIncreaseSupportedOSPlatformVersion Condition="'$(SupportedOSPlatformVersion)' != '' and $([MSBuild]::VersionGreaterThan('$(_MauiMinimumSupportedOSPlatformVersion)', '$(SupportedOSPlatformVersion)'))">true</_MauiHadToIncreaseSupportedOSPlatformVersion>
+    <_MauiHadToIncreaseTargetPlatformMinVersion   Condition="'$(TargetPlatformMinVersion)'  !=  '' and $([MSBuild]::VersionGreaterThan('$(_MauiMinimumSupportedOSPlatformVersion)', '$(TargetPlatformMinVersion)')) and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">true</_MauiHadToIncreaseTargetPlatformMinVersion>
+    <!-- Override the version numbers if they are too low. -->
+    <SupportedOSPlatformVersion Condition="'$(_MauiHadToIncreaseSupportedOSPlatformVersion)' == 'true'">$(_MauiMinimumSupportedOSPlatformVersion)</SupportedOSPlatformVersion>
+    <TargetPlatformMinVersion   Condition="'$(_MauiHadToIncreaseTargetPlatformMinVersion)' == 'true' and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">$(_MauiMinimumSupportedOSPlatformVersion)</TargetPlatformMinVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetFramework)' != '' and '$(MauiDisableSupportedOSPlatformVersionValidation)' != 'true' and '$(MauiDisableImplicitSupportedOSPlatformVersion)' != 'true'">
+    <!-- Specify the version numbers if they are missing. -->
+    <SupportedOSPlatformVersion Condition="'$(_MauiHadToIncreaseSupportedOSPlatformVersion)' != 'true' and '$(SupportedOSPlatformVersion)' == ''">$(_MauiSupportedOSPlatformVersion)</SupportedOSPlatformVersion>
+    <TargetPlatformMinVersion   Condition="'$(_MauiHadToIncreaseTargetPlatformMinVersion)' != 'true'   and '$(TargetPlatformMinVersion)'  ==  '' and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">$(_MauiSupportedOSPlatformVersion)</TargetPlatformMinVersion>
+  </PropertyGroup>
```

Comment:
This logic detects missing or old versions and will set/update. The update only happens if the MAUI SDK has a higher version that what the csproj sets - but as of today it is the minimum so nothing will actually change.